### DISCUSSION TITLE
jspsych hangs while executing a long and complex function

### DISCUSSION THREAD

**Discussion by ShengjieXu667** [jspsych/jsPsych#1221] [Q&A] [answered]
Hi all,

I am using jspsych for my online task with a high demand of randomization. I wrote a function to achieve randomizations required for my task and put it in on_finish section of my welcome page. But when I ran the task file, the whole thing hanged at the point of my function, i.e., in welcome page, one can press space key to start, but in my case, when I pressed space key, it didn't start. If I comment out the function, everything worked again. I guess it is because my function is too complex, which causes a long time of execution. Is there any suggestion for this issue? Thanks in advance!

Best,
Shengjie

**Comment by jodeleeuw** [comment, marked as answer]
Hi @ShengjieXu667 ,

A function with a long execution time will lock the browser while it runs. If the long run time is necessary, then you could try caching the results of the function. Can you run this function offline, store the results for many unique randomizations in a JSON file, and then load the results into the experiment? Then the randomization process would just be to pick one of the possible sets.

**Reply by ShengjieXu667** [reply]
Thanks for your swift reply! Yes, it's indeed a good idea to store the output of randomizations, and then read them in the code.
